scripts/combine/setupCombine.py

```diff
@@ -2330,23 +2325,35 @@ def outputFolderName(outfolder, datagroups, doStatOnly, postfix):
         add_bin_by_bin_stat_to_data_cov=args.addMCStatToCovariance,
     )
 
-    if args.fitresult:
+    if args.fitresult is not None:
         # set data from external fitresult file
         if len(args.inputFile) > 1:
             logger.warning(
                 "Theoryfit for more than one channels is currently experimental"
             )
         fitresult, fitresult_meta = combinetf2.io_tools.get_fitresult(
-            args.fitresult, meta=True
-        )
-        fitresult_data_dict, fitresult_data_cov = (
-            combinetf2.io_tools.get_fitresult_data(fitresult)
+            args.fitresult[0], meta=True
         )
 
-        writer.add_data_covariance(
-            fitresult_data_cov.get(),
+        if len(args.fitresult) > 1:
```

Comment:
what is this code doing?  ie what is the significance of fit results beyond the first one here?

Reply:
The fitresult is used now as '--fitresult <fitresult file.hdf5> <channel 1> <channel 2> ...' with an arbitrary number of channels. 

If no channels are specified the full prediction histogram and the full covariance will be taken. When specifying channels it picks the part of the covariance across these channels and the corresponding prediction histograms. 

This is needed when running the fit with masked channels and the covariance matrix is calculated across all channels, but usually only the masked channels are of interest.

Reply:
Ok I still don't understand the >1 part.  What if you specified a single channel?

Reply:
ahh ok got it